openhtf/exe/__init__.py

```diff
@@ -154,17 +89,95 @@ def Wait(self):
     """Waits until death."""
     self.join(365*24*60*60)  # Timeout needed for SIGINT handling, so 1 year.
 
-  def _ExecuteTest(self, phase_executor):
+  def GetState(self):
+    """Return the current TestState object."""
+    return self._test_state
+
+  def _ThreadProc(self):
+    """Main entry point for this Thread, conditionally loop the test."""
+    if not self.test.loop:
+      self._RunTestLoop.once(self)
+    else:
+      self._RunTestLoop()
+
+  @threads.Loop
+  def _RunTestLoop(self):
+    """Handles one whole test from start to finish.
+
+    When this finishes, the parent loops back around and calls us again.
```

Comment:
Done.  Updated to include the semantics of @threads.Loop to save someone the extra level of indirection in doc-hunting.